### PR TITLE
fix issue with mqtt concurrent map write

### PR DESCRIPTION
**Pull request by ssoroka**
closes https://github.com/influxdata/telegraf/issues/7581

### CODE REVIEW

File: plugins/inputs/mqtt_consumer/mqtt_consumer.go [hunk 1]
```diff
@@ -258,13 +262,17 @@ func (m *MQTTConsumer) recvMessage(c mqtt.Client, msg mqtt.Message) {
...
 			_, ok := m.messages[track.ID()]
+			m.messagesMutex.RUnlock()
 			if !ok {
 				// Added by a previous connection
 				continue
 			}
 			// No ack, MQTT does not support durable handling
+			m.messagesMutex.Lock()
 			delete(m.messages, track.ID())
+			m.messagesMutex.Unlock()
```
Comment (srebhan) [inline plugins/inputs/mqtt_consumer/mqtt_consumer.go:275]: Doing
```suggestion
			m.messagesMutex.RLock()
			_, ok := m.messages[track.ID()]
			if ok {
				// No ack, MQTT does not support durable handling
				delete(m.messages, track.ID())
			}
			m.messagesMutex.Unlock()
```
might also work but is holding the mutex a bit longer... On the other hand it prevents that exactly this ID is deleted in the meantime...

**Review by srebhan — APPROVED** [review]
One suggestion but the code looks good!

File: plugins/inputs/mqtt_consumer/mqtt_consumer_test.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/eclipse/paho.mqtt.golang"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
```
Comment (srebhan) [inline plugins/inputs/mqtt_consumer/mqtt_consumer_test.go:7]: Could you please elaborate on why this is necessary!?

**Review by srebhan — APPROVED** [review]
Looks good now, but I'm not sure why you alias the mqtt library in the test...

File: plugins/inputs/mqtt_consumer/mqtt_consumer_test.go (hunk 2 above)
Comment (ssoroka) [reply plugins/inputs/mqtt_consumer/mqtt_consumer_test.go:7]: this was automatically added by the formatting library, (I think goreturns, in this case). It doesn't like the implicit import of the mqtt package via a path that doesn't match the package name.

Comment (srebhan) [reply plugins/inputs/mqtt_consumer/mqtt_consumer_test.go:7]: Ok so this was my "TIL". :-) Thanks for the explanation!

**Pull request merged**